src/pybind/mgr/influx/module.py

```diff
@@ -265,70 +342,74 @@ def send_to_influx(self):
                     'detail': ['Configuration option hostname not set']
                 }
             })
-            return
+            return False
+
+        self.set_health_checks(dict())
 
         # If influx server has authentication turned off then
         # missing username/password is valid.
         self.log.debug("Sending data to Influx host: %s",
                        self.config['hostname'])
-        client = InfluxDBClient(self.config['hostname'], self.config['port'],
-                                self.config['username'],
-                                self.config['password'],
-                                self.config['database'],
-                                self.config['ssl'],
-                                self.config['verify_ssl'])
-
-        # using influx client get_list_database requires admin privs,
-        # instead we'll catch the not found exception and inform the user if
-        # db can not be created
+        database = self.config['database']
+        self.client = InfluxDBClient(self.config['hostname'],
```

Comment:
This will result in the worker threads using a mixture of the prior InfluxDBClient and the new one constructed on this line -- a bit weird.  Now that the client is accessed from multiple threads like this, it would be good to only re-do the client setup if the config has changed